tests/unit/s2n_ktls_test.c

```diff
@@ -18,45 +18,90 @@
 #include "s2n_test.h"
 #include "testlib/s2n_testlib.h"
 
-S2N_RESULT s2n_ktls_retrieve_file_descriptor(struct s2n_connection *conn, s2n_ktls_mode ktls_mode, int *fd);
-
-/* set kTLS supported cipher */
-struct s2n_cipher ktls_temp_supported_cipher = {
-    .ktls_supported = true,
-};
-struct s2n_record_algorithm ktls_temp_supported_record_alg = {
-    .cipher = &ktls_temp_supported_cipher,
-};
-struct s2n_cipher_suite ktls_temp_supported_cipher_suite = {
-    .record_alg = &ktls_temp_supported_record_alg,
-};
-
-S2N_RESULT s2n_test_configure_connection_for_ktls(struct s2n_connection *conn)
+#define SEND_FD 66
+#define RECV_FD 55
+
+static int s2n_test_setsockopt_noop(int fd, int level, int optname, const void *optval, socklen_t optlen)
+{
+    return S2N_SUCCESS;
+}
+
+static int s2n_test_setsockopt_tx(int fd, int level, int optname, const void *optval, socklen_t optlen)
+{
```

Comment:
So because you control this version of setsockopt, you could also make it take custom context the way conn->send and conn->recv do. That way you could set the expected values on the config too. That could also give you more flexibility to interrogate tls12_crypto_info_aes_gcm_128.

Reply:
So its really difficult and convoluted to set the key material on the connection. Its true that tls12_crypto_info_aes_gcm_128 is not being tested but we are going to test that in the self talk tests later.

This might not be ok and in that case I think we should test that function directly, reverting the deleted test: https://github.com/aws/s2n-tls/pull/4071/commits/77a888ca5c450dd71ddde5301d15bd96ce5c484a#diff-336cd7d2a320e19a4b7d0752555b0b6cf70e8e9f347eda4b0d2baefba79f918eL277